### DISCUSSION TITLE
Run in JupyterLite (Post API)

### DISCUSSION THREAD

**Discussion by laynr** [jupyterlite/jupyterlite#576] [Q&A]
Is it, or will it ever be, possible for JupyterLite to replicate the POST endpoint like https://jupyter-server.readthedocs.io/en/latest/developers/rest-api.html ?  This enables creating a "Run in Jupyter" button that can push a notebook to a jupyter server.

I have tried sending a POST to ./JupyterLite/Post/ and ./JupyterLite/API/Post/ to no avail.
I see https://jupyterlite.readthedocs.io/en/latest/api/ts/modules/server_src.Router.html but not sure what to do with that.

Thank you

**Comment by jtpio** [comment]
@laynr in JupyterLite the "Jupyter Server" also runs in the browser. So there is no API endpoint on the server used for serving the static files.

The POST endpoint for the contents is implemented here: 

https://github.com/jupyterlite/jupyterlite/blob/272ed62b3f8006d8c4bbc9b6699cc020e37b1268/packages/server-extension/src/index.ts#L142-L156

> This enables creating a "Run in Jupyter" button that can push a notebook to a jupyter server.

Maybe for that particular use case this issue would be relevant? https://github.com/jupyterlite/jupyterlite/issues/430
